declarations/javascript/JavascriptModulesPlugin.d.ts

```diff
@@ -0,0 +1,172 @@
+     */
+    renderModule(module: import("../Module"), renderContext: RenderContext, hooks: CompilationHooks, factory: boolean | "strict"): import("webpack-sources").Source;
+    /**
+     * @param {RenderContext} renderContext the render context
+     * @param {CompilationHooks} hooks hooks
+     * @returns {Source} the rendered source
+     */
+    renderChunk(renderContext: RenderContext, hooks: CompilationHooks): import("webpack-sources").Source;
+    /**
+     * @param {MainRenderContext} renderContext options object
+     * @param {CompilationHooks} hooks hooks
+     * @returns {Source} the newly generated source from rendering
+     */
+    renderMain(renderContext: MainRenderContext, hooks: CompilationHooks): import("webpack-sources").Source;
+    /**
+     * @param {RenderBootstrapContext} renderContext options object
+     * @param {CompilationHooks} hooks hooks
+     * @returns {{ header: string[], startup: string[], allowInlineStartup: boolean }} the generated source of the bootstrap code
+     */
+    renderBootstrap(renderContext: RenderBootstrapContext, hooks: CompilationHooks): {
+        header: string[];
+        startup: string[];
+        allowInlineStartup: boolean;
+    };
+    /**
+     * @param {RenderBootstrapContext} renderContext options object
+     * @param {CompilationHooks} hooks hooks
+     * @returns {string} the generated source of the require function
+     */
+    renderRequire(renderContext: RenderBootstrapContext, hooks: CompilationHooks): string;
+}
+declare namespace JavascriptModulesPlugin {
+    export { getCompilationHooks, chunkHasJs, Source, Chunk, ChunkGraph, ChunkHashContext, Compiler, DependencyTemplates, Module, CodeGenerationResult, ModuleGraph, RuntimeTemplate, Hash, RenderContext, MainRenderContext, RenderBootstrapContext, CompilationHooks };
+}
+type RenderContext = {
+    /**
+     * the chunk
+     */
+    chunk: import("../Chunk");
+    /**
+     * the dependency templates
+     */
+    dependencyTemplates: import("../DependencyTemplates");
+    /**
+     * the runtime template
+     */
+    runtimeTemplate: import("../RuntimeTemplate");
+    /**
+     * the module graph
+     */
+    moduleGraph: import("../ModuleGraph");
+    /**
+     * the chunk graph
+     */
+    chunkGraph: import("../ChunkGraph");
+    /**
+     * results of code generation
+     */
+    codeGenerationResults: Map<import("../Module"), import("../Module").CodeGenerationResult>;
+};
+type CompilationHooks = {
+    renderModuleContent: import("tapable").SyncWaterfallHook<[import("webpack-sources").Source, import("../Module"), RenderContext]>;
+    renderModuleContainer: import("tapable").SyncWaterfallHook<[import("webpack-sources").Source, import("../Module"), RenderContext]>;
+    renderModulePackage: import("tapable").SyncWaterfallHook<[import("webpack-sources").Source, import("../Module"), RenderContext]>;
+    renderChunk: import("tapable").SyncWaterfallHook<[import("webpack-sources").Source, RenderContext]>;
```

Comment:
The generated code could be less noisy if the `import(...)` won't be repeated in every type.

Typescript issue.